lib/diff-array.js

```diff
@@ -191,6 +200,9 @@ var handleDeepProperties = function(object, paths, cb) {
     var currentObject = object;
 
     for (var i = 0; i < props.length - 1; i++) {
+      if(_.isUndefined(currentObject[props[i]])){
```

Comment:
how do you mean? My first try was only the EJSON part in line 138 but than I had still an error that the currentObject was undefined when it tried to apply the diff changes
you can see it when you run the spec and incrementally add the changes